code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -163,7 +163,106 @@ func (ssc *StorageSmartContract) GetBlobbersHandler(
 	return sns, nil
 }
 
-func (msc *StorageSmartContract) GetTransactionByHashHandler(
+// GetAllocationBlobbersHandler returns list of all blobbers alive (e.g. excluding
+// blobbers with zero capacity).
+func (ssc *StorageSmartContract) GetAllocationBlobbersHandler(
+	ctx context.Context,
+	params url.Values, balances cstate.StateContextI) (interface{}, error) {
+
+	var creationDate = common.Timestamp(time.Now().Unix())
+
+	allocData := params.Get("allocation_data")
+	var request newAllocationRequest
+	if err := request.decode([]byte(allocData)); err != nil {
+		return "", common.NewErrInternal("can't decode allocation request", err.Error())
+	}
+
+	var sa = request.storageAllocation()
+	if balances.GetEventDB() == nil {
+		return nil, errors.New("events db is not initialised")
+	}
+	var dur = common.ToTime(sa.Expiration).Sub(common.ToTime(creationDate))
+
+	blobberIDs, err := balances.GetEventDB().GetBlobbersFromParams(paramsToMap(params), dur)
+	if err != nil || len(blobberIDs) == 0 {
+		return nil, fmt.Errorf("no blobbers found %v", err)
+	}
+
+	var size = sa.DataShards + sa.ParityShards
+
+	if len(blobberIDs) < size {
+		return nil, errors.New("not enough blobbers to honor the allocation")
+	}
+
+	return blobberIDs, nil
+}
+
+//// GetAllocationDeprecatedBlobbersHandler returns list of all blobbers alive (e.g. excluding
+//// blobbers with zero capacity).
+//func (ssc *StorageSmartContract) GetAllocationDeprecatedBlobbersHandler(
+//	ctx context.Context,
+//	params url.Values, balances cstate.StateContextI) (interface{}, error) {
```

Comment:
@Sriep Do we need this deprecated function now? Since the filterBlobbers would be redundant once we start moving to eventsDB, although I have kept it as of now if we need to rely on the same.

Reply:
If this GetAllocationDeprecatedBlobbersHandler is not being used, we can remove it.